tests/pam_duo-4.t

```diff
@@ -43,32 +43,36 @@ Preauth states
   [4] Failsafe Duo login for 'preauth-ok-missing_response': BSON missing valid 'response'
   [4] Failsafe Duo login for 'preauth-fail-missing_response': BSON missing valid 'code'
   [4] Failsafe Duo login for 'preauth-bad-stat'
-  [4] Failsafe Duo login for 'preauth-fail': BSON missing valid 'response'
-  [4] Aborted Duo login for 'preauth-deny': you suck
-  you suck
-  [4] Skipped Duo login for 'preauth-allow': you rock
+  [4] Failed Duo login for 'preauth-fail': 1000: Pre-authentication failed
```

Comment:
this particular test sticks out to me as odd.
Since all we were doing is making name changes I'm confused why the response is now printing 3 times instead of just once.
Do you happen to know why this is happening?

Reply:
Yes, @xdesai take a look [here](https://github.com/duosecurity/duo_unix/pull/161/commits/9be5b24d5a3012ad6ec1e23e18a3a02e37fbaebb#diff-856049f897cb7e6ea2fe0e183d3b4d7bL148).  Previously we were not setting the ret variable correctly.  Also, when pam_duo fails, it doesn't hit a break statement in [this loop](https://github.com/duosecurity/duo_unix/blob/master/pam_duo/pam_duo.c#L268-L279), so it continues to loop.

Reply:
ahhh I see now! thanks for clarifying